### PR TITLE
Add JULIA_PKG_GC_AUTO to docs

### PR DESCRIPTION
**Pull request by pfitzseb**
Documentation for https://github.com/JuliaLang/Pkg.jl/pull/3642.

### CODE REVIEW

**Comment by fingolfin** [comment]
This feature unfortunately still is not in Pkg.jl despite https://github.com/JuliaLang/Pkg.jl/pull/3642 being approved in October 2023.

At the same time, this PR now has a merge conflict :-(

File: doc/src/manual/environment-variables.md [hunk 1]
```diff
@@ -267,6 +267,14 @@ versions of packages already installed as possible.
 !!! compat "Julia 1.9"
     This only affects Julia 1.9 and above.
 
+### [`JULIA_PKG_GC_AUTO`](@id JULIA_PKG_GC_AUTO)
+
+If set to `false`, automatic garbage collection of packages and artifacts will be disabled;
+see [`Pkg.gc`](https://pkgdocs.julialang.org/v1/api/#Pkg.gc) for more details.
+
+!!! compat "Julia 1.12"
```
Comment (pfitzseb) [inline doc/src/manual/environment-variables.md:275]: I assume this won't be in 1.11.

**Review by IanButterworth — APPROVED** [review]

**Comment by IanButterworth** [comment]
Let's wait for a Pkg bump later today before merging

**Comment by lgoettgens** [comment]
> Let's wait for a Pkg bump later today before merging

The Pkg bump was https://github.com/JuliaLang/julia/pull/55258. So this should be good to get merged now, right?

**Pull request merged**